cloudinit/sources/DataSourceAzure.py

```diff
@@ -1156,42 +1156,39 @@ def exc_cb(msg, exception):
 
                     report_ready_succeeded = self._report_ready(lease=lease)
```

Comment:
@anhvoms This [unit test](https://github.com/canonical/cloud-init/blob/d95b448fe106146b7510f7b64f2e83c51943f04d/tests/unittests/test_datasource/test_azure.py#L1484-L1485) (already in cloud-init master from previous work) ensures that if `_report_ready` fails, it will only return `False` and not raise an exception.